internal/buildpackage/builder.go

```diff
@@ -56,51 +144,71 @@ func (p *PackageBuilder) Save(repoName string, publish bool) (imgutil.Image, err
 		}
 	}
 
-	if len(stacks) == 0 {
-		return nil, errors.Errorf("no compatible stacks among provided buildpacks")
-	}
+	return stacks
+}
 
-	image, err := p.imageFactory.NewImage(repoName, !publish)
-	if err != nil {
-		return nil, errors.Wrapf(err, "creating image")
+func (b *PackageBuilder) SaveAsFile(path string) error {
+	if err := b.validate(); err != nil {
+		return err
 	}
 
-	if err := dist.SetLabel(image, MetadataLabel, &Metadata{
-		BuildpackInfo: p.buildpack.Descriptor().Info,
-		Stacks:        stacks,
-	}); err != nil {
-		return nil, err
+	layoutImage := &layoutImage{
+		Image: empty.Image,
 	}
 
 	tmpDir, err := ioutil.TempDir("", "package-buildpack")
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer os.RemoveAll(tmpDir)
 
-	bpLayers := dist.BuildpackLayers{}
-	for _, bp := range append(p.dependencies, p.buildpack) {
-		bpLayerTar, err := dist.BuildpackToLayerTar(tmpDir, bp)
-		if err != nil {
-			return nil, err
-		}
+	if err := b.finalizeImage(tmpDir, layoutImage); err != nil {
+		return err
+	}
 
-		if err := image.AddLayer(bpLayerTar); err != nil {
-			return nil, errors.Wrapf(err, "adding layer tar for buildpack %s", style.Symbol(bp.Descriptor().Info.FullName()))
-		}
+	layoutDir, err := ioutil.TempDir(tmpDir, "oci-layout")
+	if err != nil {
+		return errors.Wrap(err, "creating oci-layout temp dir")
+	}
 
-		diffID, err := dist.LayerDiffID(bpLayerTar)
-		if err != nil {
-			return nil, errors.Wrapf(err,
-				"getting content hashes for buildpack %s",
-				style.Symbol(bp.Descriptor().Info.FullName()),
-			)
-		}
+	p, err := layout.Write(layoutDir, empty.Index)
+	if err != nil {
+		return errors.Wrap(err, "writing index")
+	}
 
-		dist.AddBuildpackToLayersMD(bpLayers, bp.Descriptor(), diffID.String())
+	if err := p.AppendImage(layoutImage); err != nil {
+		return errors.Wrap(err, "writing layout")
 	}
 
-	if err := dist.SetLabel(image, dist.BuildpackLayersLabel, bpLayers); err != nil {
+	outputFile, err := os.Create(path)
+	if err != nil {
+		return errors.Wrap(err, "creating output file")
+	}
+	defer outputFile.Close()
+
+	tw := tar.NewWriter(outputFile)
+	defer tw.Close()
+
+	return archive.WriteDirToTar(tw, layoutDir, "/", 0, 0, 0644, true)
```

Comment:
This should probably be `0755` given that if a user extracts the contents, they will find it difficult to traverse the directory. There is no added necessary security concerns to be concerned about since the contents (in the layers) are archived independently.